integration_tests/run.py

```diff
@@ -0,0 +1,239 @@
+PIPELINE_NAME = 'gcp-variant-transforms-integration-test'
+SCOPES = ['https://www.googleapis.com/auth/bigquery']
+DEFAULT_ZONES = ['us-west1-b']
+SCRIPT_PATH = '/opt/gcp_variant_transforms/bin/vcf_to_bq'
+
+
+class TestCase(object):
+  """Test case that holds information to run in pipelines API.
+
+  To define a new integration test case, create a json file in
+  gcp_variant_transforms/integration_tests directory and specify at least
+  test_name, table_name, and input_pattern for the integration test.
+  """
+
+  def __init__(self,
+               context,
+               test_name,
+               table_name,
+               input_pattern,
+               **kwargs):
+
+    self._name = test_name
+    dataset_id = context.dataset_id
+    output_table = '{}:{}.{}'.format(context.project, dataset_id, table_name)
+    args = ['--input_pattern {}'.format(input_pattern),
+            '--output_table {}'.format(output_table),
+            '--project {}'.format(context.project),
+            '--staging_location {}'.format(context.staging_location),
+            '--temp_location {}'.format(context.temp_location),
+            '--job_name {}-{}'.format(test_name, dataset_id.replace('_', '-'))]
+    for k, v in kwargs.iteritems():
+      args.append('--{} {}'.format(k, v))
+
+    self._request = {
+        'pipelineArgs': {
+            'projectId': context.project,
+            'logging': {'gcsPath': context.logging_location},
+            'serviceAccount': {'scopes': SCOPES}
+        },
+        'ephemeralPipeline': {
+            'projectId': context.project,
+            'name': PIPELINE_NAME,
+            'resources': {'zones': DEFAULT_ZONES},
+            'docker': {
+                'imageName': IMAGE_NAME,
+                'cmd': SCRIPT_PATH + ' ' + ' '.join(args)
+            },
+        }
+    }
+
+  def get_name(self):
+    return self._name
+
+  def get_request(self):
+    return self._request
+
+
+class TestContextManager(object):
+  """Manages all resources for a given run of tests.
+
+  Responsible for setting up tests (i.e. creating a unique dataset) and
+  cleaning up all resources after tests have been completed.
+  """
+
+  def __init__(self, known_args):
+    self.staging_location = known_args.staging_location
+    self.temp_location = known_args.temp_location
+    self.logging_location = known_args.logging_location
+    self.project = known_args.project
+
+  def __enter__(self):
+    self.credentials = GoogleCredentials.get_application_default()
+    client = bigquery.Client(project=self.project)
+    self.dataset_id = 'integration_tests_{}'.format(
+        datetime.now().strftime('%Y%m%d_%H%M%S'))
+    dataset_ref = client.dataset(self.dataset_id)
+    self.dataset = bigquery.Dataset(dataset_ref)
+    _ = client.create_dataset(self.dataset)
+    return self
+
+  def __exit__(self, *args):
+    client = bigquery.Client(project=self.project)
+    tables = client.list_dataset_tables(self.dataset)
+    for table in tables:
```

Comment:
Please add a comment about why separately deleting tables is necessary.

Reply:
Done.